oxidation/libparsec/crates/types/src/shamir.rs

```diff
@@ -0,0 +1,102 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
+
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::{collections::HashMap, num::NonZeroU64};
+
+use libparsec_serialization_format::parsec_data;
+
+use crate::{
+    self as libparsec_types, impl_transparent_data_format_conversion, DataError, DataResult,
+    DateTime, DeviceID, UserID,
+};
+
+fn load<T: DeserializeOwned>(raw: &[u8]) -> DataResult<T> {
+    rmp_serde::from_slice(raw).map_err(|_| Box::new(DataError::Serialization))
+}
+
+fn dump<T: Serialize>(data: &T) -> Vec<u8> {
+    rmp_serde::to_vec_named(data).expect("Unreachable")
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(
+    into = "ShamirRecoveryBriefCertificateData",
+    from = "ShamirRecoveryBriefCertificateData"
+)]
+pub struct ShamirRecoveryBriefCertificate {
+    pub author: DeviceID,
+    pub timestamp: DateTime,
+    pub threshold: NonZeroU64,
+    pub per_recipient_shares: HashMap<UserID, NonZeroU64>,
+}
+
+impl ShamirRecoveryBriefCertificate {
+    pub fn load(raw: &[u8]) -> DataResult<Self> {
+        load(raw)
+    }
+    pub fn dump(&self) -> Vec<u8> {
+        dump(self)
+    }
+}
+
+parsec_data!("schema/shamir/shamir_recovery_brief_certificate.json5");
+
+impl_transparent_data_format_conversion!(
+    ShamirRecoveryBriefCertificate,
+    ShamirRecoveryBriefCertificateData,
+    author,
+    timestamp,
+    threshold,
+    per_recipient_shares,
+);
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(
+    into = "ShamirRecoveryShareCertificateData",
+    from = "ShamirRecoveryShareCertificateData"
+)]
+pub struct ShamirRecoveryShareCertificate {
+    pub author: DeviceID,
+    pub timestamp: DateTime,
+    pub recipient: UserID,
+    pub ciphered_share: ShamirRecoveryShareData,
```

Comment:
`ShamirRecoveryShareData` don't hold sensitive information ?

Reply:
It is, but it's ciphered now.